File: medium/sort-colors.py
from collections import OrderedDict

class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        counts = OrderedDict()
        counts[0] =0
        counts[1] =0
        counts[2] =0 
        
        for n in nums:
            counts[n] +=1
            
        print(counts)
        r = [] 
        i = 0 
        for c in counts: 
            for _ in range(counts[c]):
                nums[i] = c
                i +=1

File: medium/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int ones =0; 
        int twos =0; 
        
        for(int ele : nums){
            if (ele ==1){
                ones ++;
            } else if (ele ==2){
                twos ++;
            }
        }
        
        for (int i = nums.length-1; i>=0; i--){
            if (twos > 0){
                nums[i] =2;
                twos --;
            } else if (ones >0){
                nums[i]=1;
                ones--;
            } else {
                nums[i] =0;
            }
        }
    }
}
